### PR TITLE
[EH] add possibility to use custom filters

### PR DESCRIPTION
**Pull request by tswonke**
see [https://github.com/Laravel-Backpack/CRUD/issues/1385](https://github.com/Laravel-Backpack/CRUD/issues/1385)

### CODE REVIEW

File: src/PanelTraits/Filters.php [hunk 1]
```diff
@@ -215,13 +215,22 @@ public function __construct($options, $values, $filter_logic)
...
 
         if (\Request::has($this->name)) {
             $this->currentValue = \Request::input($this->name);
         }
     }
 
+    public function setView()
+    {
+        if (isset($this->options['view'])) {
+            $this->view = $this->options['view'] . '.' . $this->type;
```
Comment (OliverZiegler) [inline src/PanelTraits/Filters.php]: Should this not be the exact view without `$this->type` suffix?
Seems like bad naming in the current case... 🤔

File: src/PanelTraits/Filters.php [hunk 2]
```diff
@@ -215,13 +215,22 @@ public function __construct($options, $values, $filter_logic)
 
         $this->values = $values;
         $this->options = $options;
-        $this->view = 'crud::filters.'.$this->type;
+        $this->setView();
```
Comment (OliverZiegler) [inline src/PanelTraits/Filters.php]: Better don't rely on `$this->options` already set, but add `$options` as a parameter.

File: src/PanelTraits/Filters.php [hunk 3]
```diff
@@ -215,13 +215,22 @@ public function __construct($options, $values, $filter_logic)
...
 
         if (\Request::has($this->name)) {
             $this->currentValue = \Request::input($this->name);
         }
     }
 
+    public function setView()
```
Comment (OliverZiegler) [inline src/PanelTraits/Filters.php]: See above for `$options` parameter.

**Comment by tabacitu** [comment]
Thank you @tswonke - improved on it a little bit and merged (we can now use the null coalesce operator since we force PHP 7).

Thanks a lot!

**Pull request merged**